src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -142,9 +162,9 @@ private static void writeReadsSingle(
             return new Tuple2<>(gatkRead, samRecordWritable);
         });
 
-        // coalesce is needed for writing to a single bam.
+        // do a total sort so that all the reads in partition i are less than those in partition i+1
         final JavaPairRDD<GATKRead, SAMRecordWritable> out =
-                rddSamRecordWriteable.sortByKey(new ReadCoordinateComparator(header)).coalesce(1);
+                totalSortByKey(rddSamRecordWriteable, new ReadCoordinateComparator(header), GATKRead.class);
```

Comment:
@laserson I think you are right - not sure how I missed that Spark does this for you! I've submitted https://github.com/broadinstitute/gatk/pull/1363 to simplify the code.